main.js

```diff
@@ -53,57 +66,103 @@ app.on('activate', () => {
 // In this file you can include the rest of your app's specific main process
 // code. You can also put them in separate files and require them here.
 
-let configObj
-let installedAddonsObj // Dictonary of all installed addons. Reference addons using "name" as key
+// --- Initialization Start---
+let configObj // JSON object that holds application config such as location of addon directory and installed addons file
+let installedAddonsDict // Dictionary of all installed addons. Reference addons using "name" as key
 
-initConfig().then((value) => { 
-  configObj = value
-  return value
-}).then((value) => {
-  initAddonList(value).then((value) => { 
-    installedAddonsObj = value})
-})
+initConfig()
+  .then(value => {
+    configObj = value // Sets config settings
+    return configObj
+  })
+  .then(configObj => {
+    readAddonList(configObj).then(val => {
+      if (configObj.checkUpdateOnStart === true) {
+        checkAllUpdates(val)
+      }
+    })
+    
+    const installedAddonsJsonWatcher = chokidar.watch(configObj.addonRecordFile, { persistent: true }) // Watches for changes in addons.json,
```

Comment:
This looks better, we may want to create a class wrapper for this watcher in the future or remove the watcher since chokidar seems to typically be used for hot reloading, or files that are constantly changing. (***Don't change it now this works for the time being***)